dags/transportation/dockless/geom_parse.sql

```diff
@@ -0,0 +1,72 @@
+/* create a trip_geoms table */ 
+CREATE TABLE IF NOT EXISTS trips_geoms (
+	trip_id UUID NOT NULL,
+	points GEOMETRY 
+); 
+/* process route function */ 
+CREATE OR REPLACE FUNCTION process_route(jsonb) 
+	RETURNS geometry AS 
+	$func$
+	BEGIN
+		RETURN ST_MakeLine(ARRAY(
+					SELECT ST_GeomFromGeoJSON((feature -> 'geometry')::text) as geometry FROM (
+						SELECT jsonb_array_elements($1 -> 'features') as feature
+					) features
+				  ));
+	EXCEPTION WHEN OTHERS THEN
+		return NULL;
+	END
+	$func$  
+    LANGUAGE plpgsql IMMUTABLE;
+
+/* trigger function */ 
+CREATE OR REPLACE FUNCTION geom_copy() RETURNS TRIGGER AS
+$BODY$
+BEGIN
+    INSERT INTO
+        trips_geoms(trip_id,points)
+        VALUES(new.trip_id,process_route(new.route));
+           RETURN new;
+END;
+$BODY$
+language plpgsql;
+/* trigger itself */ 
+CREATE OR REPLACE TRIGGER route_process
+     AFTER INSERT ON trips
+     FOR EACH ROW
+     EXECUTE PROCEDURE geom_copy();
+
+/* create a status_change_geoms table */ 
+CREATE TABLE IF NOT EXISTS status_change_geoms (
+	status_change_id INTEGER NOT NULL REFERENCES status_changes(id),
+	event_location_geom GEOMETRY 
+); 
+/* process location function function */ 
+-- jsonb input
+CREATE OR REPLACE FUNCTION process_status_change_location(jsonb)
+    	RETURNS geometry AS 
+	$func$
+	BEGIN
+		RETURN st_setsrid(st_geomfromgeojson(($1 -> 'geometry'::text)::text), 4326);
+	EXCEPTION WHEN OTHERS THEN
+		return NULL;
+	END
+	$func$  
+    LANGUAGE plpgsql IMMUTABLE;
+
+/* trigger function */ 
+CREATE OR REPLACE FUNCTION status_geom_copy() RETURNS TRIGGER AS
+$BODY$
+BEGIN
+    INSERT INTO
+        status_change_geoms(status_change_id, event_location_geom)
```

Comment:
As long as we're here, should we add any of the other fields to the `trips_geom` and `status_change_geom` tables? Is that data going to break the bank?

Reply:
status_change_geoms and trips_geoms get joined to the original tables in the materialized view update (see #72).

Reply:
Ah, makes sense.